packages/okta-react-native/android/src/main/java/com/oktareactnative/OktaSdkBridgeModule.java

```diff
@@ -131,12 +145,54 @@ public void signIn() {
     }
 
     @ReactMethod
-    public void authenticate(String sessionToken) {
+    public void signIn(String username, String password, final Promise promise) {
+        try {
+            authnClient.authenticate(username, password.toCharArray(), null, new AuthenticationStateHandlerAdapter() {
+                @Override
+                public void handleUnknown(AuthenticationResponse authenticationResponse) {
+                    WritableMap params = Arguments.createMap();
+                    params.putString(OktaSdkConstant.ERROR_CODE_KEY, OktaSdkError.SIGN_IN_FAILED.getErrorCode());
+                    params.putString(OktaSdkConstant.ERROR_MSG_KEY, OktaSdkError.SIGN_IN_FAILED.getErrorMessage());
+                    sendEvent(reactContext, OktaSdkConstant.ON_ERROR, params);
+
+                    promise.reject(OktaSdkError.SIGN_IN_FAILED.getErrorCode(), OktaSdkError.SIGN_IN_FAILED.getErrorMessage());
```

Comment:
I'm not sure we should be wrapping authn-java with the react-native sdk. The reason is that this code segment does not cover all the cases of authn API. If developers run into issues then they would have to edit this module and add the correct handling. So they would have to modify the SDK itself to fix these issues. This is only used as a sample in the android oidc-sdk because it is in the app layer and developers are responsible for error cases. This puts the responsibility in the sdk level.